src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java

```diff
@@ -392,6 +394,54 @@ void testIsCaseUrgent(String value) {
         assertThat(result).isFalse();
     }
 
+
+    @Test
+    void getHearingLocations_shouldReturnCorrespondingEpimsIdForVenue() throws URISyntaxException, IOException {
+        SscsCaseData caseData = SscsCaseData.builder()
+            .appeal(Appeal.builder()
+                        .hearingOptions(HearingOptions.builder().build())
+                        .build())
+            .processingVenue(PROCESSING_VENUE_1)
+            .build();
+
+        given(venueService.getEpimsIdForVenue(caseData.getProcessingVenue())).willReturn(Optional.of("9876"));
+        given(referenceDataServiceHolder.getVenueService()).willReturn(venueService);
+
+        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(
+            caseData,
+            referenceDataServiceHolder
+        );
+
+        assertThat(result).hasSize(1);
+        assertThat(result.get(0).getLocationId()).isEqualTo("9876");
+        assertThat(result.get(0).getLocationType()).isEqualTo(COURT);
+    }
+
+    @DisplayName("Multiple hearing location Test")
+    @Test
+    void getMultipleHearingLocations_shouldReturnCorrespondingMultipleEpimsIdForVenue() throws URISyntaxException, IOException {
+        SscsCaseData caseData = SscsCaseData.builder()
+            .appeal(Appeal.builder()
+                        .hearingOptions(HearingOptions.builder().build())
+                        .build())
+            .processingVenue(PROCESSING_VENUE_1)
+            .build();
+
+        given(venueService.getEpimsIdForVenue(caseData.getProcessingVenue())).willReturn(Optional.of("443014"));
+        given(referenceDataServiceHolder.getVenueService()).willReturn(venueService);
+
+        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(
+            caseData,
+            referenceDataServiceHolder
+        );
+
+        assertThat(result).hasSize(2);
+        assertThat(result.get(0).getLocationId()).isEqualTo("226511");
```

Comment:
I personally would not tie the test to the actual json file, I'd mock everything out

Reply:
Currently hard to do due to the architecture of the classes. We'll put it up as technical debt and do it as part of a separate ticket.